flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGenerator.java

```diff
@@ -591,6 +593,13 @@ private void connect(Integer headOfChain, StreamEdge edge) {
 		}
 	}
 
+	private void checkCompatible(ResultPartitionType type, long bufferTimeout) {
+		if (type.isBlocking() && bufferTimeout != -1) {
+			throw new UnsupportedOperationException("Blocking partition does not support buffer timeout at the" +
```

Comment:
I think adding the timeout value and `edge.toString` could help to find out misconfguration to the user.

Reply:
Agree, i will refactor the message.